src/app/AttributeCache.h

```diff
@@ -311,7 +326,7 @@ class AttributeCache : protected ReadClient::Callback
     }
 
 private:
-    using AttributeState = Variant<System::PacketBufferHandle, StatusIB>;
+    using AttributeState = Variant<VersionedAttributeBuffer, StatusIB>;
```

Comment:
I missed this in my earlier reviews: -Data version is unique per cluster, not per attribute. So we shouldn't be modeling it like this. Rather, we should alter `ClusterState` to be a complex struct:

```
struct ClusterState {
    DataVersion dataVersion;
    std::map<AttributeId, AttributeState> attributeState;
}
```

Reply:
Dis-regard this comment given the comment above.